docs/book/en/src/frameworks/mesh/wasm-nginx-module.md

```diff
@@ -0,0 +1,137 @@
+# wasm-nginx-module
+
+wasm-nginx-module is an Nginx module that is built upon OpenResty. By implementing the [Proxy Wasm ABI](https://github.com/proxy-wasm/spec), any Wasm program written with Proxy Wasm SDK can be run inside it. Hence, you can write Go or Rust code, compile them into Wasm, then load & execute it in the Nginx.
+
+The wasm-nginx-module is already used in APISIX and allows it to [run Wasm plugin like Lua plugin](https://github.com/apache/apisix/blob/master/docs/en/latest/wasm.md).
+
+Let me show you a real-world example - use Wasm to inject custom responses in Nginx.
+
+## Inject Custom Response via Go in Nginx, Step by Step
+
+### Step 1: Write code based on proxy-wasm-go-sdk
+
+The implementation code (including go.mod and others) can be found at [here](https://github.com/apache/apisix/tree/master/t/wasm).
+
+It should be explained that although the proxy-wasm-go-sdk project carries the Go name, it actually uses tinygo instead of native Go, which has some problems supporting WASI (which you can think of as a non-browser WASM runtime interface), see [here](https://github.com/tetratelabs/proxy-wasm-go-sdk/blob/main/doc/OVERVIEW.md#tinygo-vs-the-official-go-compiler) for more details.
+
+We also provide a Rust version (including Cargo.toml and others) [there](https://github.com/api7/wasm-nginx-module/tree/main/t/testdata/rust/fault-injection).
+
+### Step 2: Build the corresponding Wasm file
+
+```shell
+tinygo build -o ./fault-injection/main.go.wasm -scheduler=none -target=wasi ./fault-injection/main.go
+```
+
+### Step 3: Load and execute the Wasm file
+
+Build your Nginx with wasm-nginx-module included and WasmEdge shared library installed in the right path.
```

Comment:
Can you provide a link or a paragraph on how to do this exactly? Do APISIX binary distributions have this already? Thanks!

Reply:
There is a section about setting up WasmEdge when building Nginx: https://github.com/api7/wasm-nginx-module#install-dependencies.

However, it is more for developers, which requires the reader to have Nginx C module development experience.

Reply:
I made some changes. Please let me know if you agree. Thanks.

Reply:
LGTM